### ISSUE TITLE
Error building msvc_15

### ISSUE THREAD

**Issue by AndreyMlashkin** [conan-io/conan-docker-tools#226]
### Environment Details
* Conan Docker Tools image: **msvc_15**
* Operating System: **Windows**
* Operation System Version: **Windows 10**
* Python version (If you are building some image): **python 3.7.0**

### Steps to reproduce (Include if Applicable)
cd conan-docker-tools\msvc_15
docker build . --no-cache

### Build logs (Include if Available)

Sending build context to Docker daemon  4.608kB
Step 1/11 : FROM microsoft/windowsservercore:ltsc2016
 ---> f9a2491293f2
Step 2/11 : LABEL maintainer="Luis Martinez de Bartolome <luism@jfrog.com>"
 ---> Running in 7a490f351def
Removing intermediate container 7a490f351def
 ---> 7f5e95549b00
Step 3/11 : SHELL ["powershell.exe", "-ExecutionPolicy", "Bypass", "-Command"]
 ---> Running in c7e7ab3e01d7
Removing intermediate container c7e7ab3e01d7
 ---> e6c3097f2668
Step 4/11 : ENV chocolateyUseWindowsCompression=false     PYTHONIOENCODING=UTF-8
 ---> Running in 0c41ad3147f3
Removing intermediate container 0c41ad3147f3
 ---> 35f9f8a6fb5c
Step 5/11 : RUN iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'));     $env:Path += '";C:\tools\python3;C:\tools\python3\Scripts"';     choco install --no-progress --yes git --version=2.19.0 --params '"/InstallDir:C:\tools\git"';     choco install --no-progress --yes svn --version=1.8.17 --params '"/InstallDir:C:\tools\svn"';     choco install --no-progress --yes cmake --version=3.12.2 --params '"/InstallDir:C:\tools\cmake"' --installargs 'ADD_CMAKE_TO_PATH=""System""';     choco install --no-progress --yes python3 --version=3.7.0 --params '"/InstallDir:C:\tools\python3"'
 ---> Running in ddaf14d7eaf2
Exception calling "DownloadString" with "1" argument(s): "The request was
aborted: Could not create SSL/TLS secure channel."
At line:1 char:1
+ iex ((New-Object System.Net.WebClient).DownloadString('https://chocol ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : WebException

choco : The term 'choco' is not recognized as the name of a cmdlet, function,
script file, or operable program. Check the spelling of the name, or if a path
was included, verify that the path is correct and try again.
At line:1 char:154
+ ... v:Path += ';C:\tools\python3;C:\tools\python3\Scripts'; choco install ...
+                                                             ~~~~~
    + CategoryInfo          : ObjectNotFound: (choco:String) [], CommandNotFou
   ndException
    + FullyQualifiedErrorId : CommandNotFoundException

choco : The term 'choco' is not recognized as the name of a cmdlet, function,
script file, or operable program. Check the spelling of the name, or if a path
was included, verify that the path is correct and try again.
At line:1 char:246
+ ... t --version=2.19.0 --params '/InstallDir:C:\tools\git'; choco install ...
+                                                             ~~~~~
    + CategoryInfo          : ObjectNotFound: (choco:String) [], CommandNotFou
   ndException
    + FullyQualifiedErrorId : CommandNotFoundException

choco : The term 'choco' is not recognized as the name of a cmdlet, function,
script file, or operable program. Check the spelling of the name, or if a path
was included, verify that the path is correct and try again.
At line:1 char:338
+ ... n --version=1.8.17 --params '/InstallDir:C:\tools\svn'; choco install ...
+                                                             ~~~~~
    + CategoryInfo          : ObjectNotFound: (choco:String) [], CommandNotFou
   ndException
    + FullyQualifiedErrorId : CommandNotFoundException

choco : The term 'choco' is not recognized as the name of a cmdlet, function,
script file, or operable program. Check the spelling of the name, or if a path
was included, verify that the path is correct and try again.
At line:1 char:475
+ ... \tools\cmake' --installargs 'ADD_CMAKE_TO_PATH=System'; choco install ...
+                                                             ~~~~~
    + CategoryInfo          : ObjectNotFound: (choco:String) [], CommandNotFou
   ndException
    + FullyQualifiedErrorId : CommandNotFoundException

The command 'powershell.exe -ExecutionPolicy Bypass -Command iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'));     $env:Path += '";C:\tools\python3;C:\tools\python3\Scripts"';     choco install --no-progress --yes git --version=2.19.0 --params '"/InstallDir:C:\tools\git"';     choco install --no-progress --yes svn --version=1.8.17 --params '"/InstallDir:C:\tools\svn"';     choco install --no-progress --yes cmake --version=3.12.2 --params '"/InstallDir:C:\tools\cmake"' --installargs 'ADD_CMAKE_TO_PATH=""System""';     choco install --no-progress --yes python3 --version=3.7.0 --params '"/InstallDir:C:\tools\python3"'' returned a non-zero code: 1

**Comment by Croydon** [comment]
@uilianries Since we don't really use the MSVC containers and we don't really test or maintain them, maybe we should remove them from the repo?

We added them once, because we were thinking that we can containerize our Windows CI environment similar to Linux. This would also have improved (local) debugging and gave more independence of the actual CI host. However, MSVC does not allow to upload the built containers anywhere public, so sharing isn't legal and building them every time takes too long. I guess, that at this point there are pretty useless to us.

**Comment by uilianries** [comment]
@Croydon agreed

**Comment by solvingj** [comment]
I agree that it's been an issue that the windows `Dockerfile`s have been maintained here but not used for Conan Center builds at all. However, the timing of this ticket and PR are unfortunate.

In the Conan team, as of a few weeks ago, we have added a task to move all windows builds to use windows containers on our CI.  Once we are building Conan Center packages with windows containers, it seems like the `Dockerfile` should be public so that users (both conan team and OSS users) can reproduce the environment if they wish to.  At that point, it seems like this will be the appropriate place for them.

So, delete them now if you want but we can always re-add them? 
Or, just leave them because we'll be updating them in the near future?

I'll leave it to @uilianries

**Comment by solvingj** [comment]
Regarding @Croydon comment:
> However, MSVC does not allow to upload the built containers anywhere public

The general recommendation for professional uses of Conan docker images would be for the build engineers in those teams to add CI jobs to build Conan CI images from source, and then upload those images to their own private docker repositories.  In the case of the windows image, it's just required.  

I think it's still very good for the community to have a working reference of "how to install conan in a windows docker container" (so long as we're actually using it ourselves).

**Comment by Croydon** [comment]
> In the Conan team, as of a few weeks ago, we have added a task to move all windows builds to use windows containers on our CI.

This is good news. If there are used and maintained they are a great addition.

**Comment by uilianries** [comment]
> In the Conan team, as of a few weeks ago, we have added a task to move all windows builds to use windows containers on our CI.

So it's totally valid keeping Windows docker images. As only few people use it, the maintenance cost is high, but if we include it to CCI, I see totally valid spending some time working on it.

**Comment by uilianries** [comment]
https://github.com/conan-io/conan-docker-tools#visual-studio

**Comment by Croydon** [comment]
What happened to the idea of using Windows containers on CCI and publishing their Dockerfiles in this repo? (no binary uploads because of license)

**Comment by uilianries** [comment]
Too complex to implement. We tried to use Docker on Windows, spent weeks trying, but had many different errors, when we solved an error, an new one happened. Installing Visual Studio community with all support was really hard (Visual Studios Tools does not have devenv.exe installed).

So we decided to use Windows machines, but with Ansible. So we can destroy a machine (they are not physical), and start a new one from zero, installing all stuff needed using a script. 

If you remember, we had problem with Visual Studio license, which was a nightmare to avoid. Now, using those new machines, we can control it better.
